### PR TITLE
Correctly reflect model owner in the URL. 

### PR DESCRIPTION
**Pull request by frankban**
Also, do not hang when switching to current model.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6693/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6694/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6695/

**Comment by frankban** [comment]
**QA** must be done in both gijoe and gisf scenarios. 
For gisf, at this stage you can use `guiproxy -config 'gisf: true, jujuEnvUUID: ""'`.

- switch to a different model from the model switcher: ensure it works and the URL reflects the new model;
- switch to the same model from the model switcher: this should be a noop;
- go to the profile view and ensure the current model is expanded;
- switch to a different model from the profile view: ensure it works and the URL reflects the new model;
- click to go to the profile view again;
- switch to the same model (should be the one already expanded): ensure it works and the URL reflects the new model;
- create a new model: ensure it works, the URL does not reflect the new model, but this is a known issue;
- switch to a model owned by another user (this is easy enough in gisf, switching to jujugui-org): ensure the URL reflects the real model owner (e.g.: `/u/uros-jovanovic/jujugui-org`).

File: jujugui/static/gui/src/app/app.js [hunk 1]
```diff
@@ -1974,10 +1985,9 @@ YUI.add('juju-gui', function(Y) {
           this._renderCharmbrowser(state, next);
           break;
         case 'new':
-          // When going to disconnected mode the modelUUID should be set to
-          // a falsy value. If it's not then we're still connected and need to
-          // switch.
-          if (this.get('modelUUID')) {
+          // When going to disconnected mode we need to be disconnected from
+          // models.
+          if (this.env.get('connected')) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/app.js]: Nice improvement. Good idea

File: jujugui/static/gui/src/app/app.js [hunk 2]
```diff
@@ -1828,9 +1833,15 @@ YUI.add('juju-gui', function(Y) {
             root: null
           });
         } else {
-          // If no model was found then put the user in disconnected mode.
+          // If no model was found then navigate to the user profile.
+          const msg = `no such charm, bundle or model: u/${modelPath}`;
+          // TODO frankban: here we should put a notification, but we can't as
```
Comment (hatched) [inline jujugui/static/gui/src/app/app.js]: This has me a little concerned, I'll add it to my list of things to investigate.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6696/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6697/

File: jujugui/static/gui/src/app/views/utils.js [hunk 3]
```diff
@@ -1302,15 +1302,21 @@ YUI.add('juju-view-utils', function(Y) {
...
+      - id: the model unique identifier;
+      - owner: the user owning the model, like "admin" or "who@external".
     @param {Boolean} confirmUncommitted Whether to show a confirmation if there
       are uncommitted changes.
   */
-  utils.switchModel = function(
-    env, uuid, name, confirmUncommitted=true) {
-    const switchModel =
-      utils._switchModel.bind(this, env, uuid, name);
+  utils.switchModel = function(env, model, confirmUncommitted=true) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/views/utils.js]: Thanks for simplifying the call signature even more

**Comment by hatched** [comment]
:shipit: Shipping to unblock other branches, any issues can be resolved in a follow-up

Thanks all

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Comment by kadams54** [comment]
QA is :ok:

**QA Notes:**

* Ran through both `gisf` and `gijoe` in both Chrome and Firefox.
* Per a comment from @frankban in IRC, I added another step to the QA instructions, to refresh on the profile page.
* I pointed my `guiproxy` to production jimm for `gisf` testing, so I used a slight modification on the `guiproxy` command listed in the QA instructions: `guiproxy -config 'gisf: true, jujuEnvUUID: ""' -controller jimm.jujucharms.com:443`.
* When I initially started `gisf` testing in Chrome, I ran into an issue where the model switcher 1) lost the profile button and 2) listed the models as `username/modelname`. I have no idea how this happened; I refreshed and tried to reproduce the steps I'd been through, but was not able to replicate the issue.
* `gijoe` was tested on AWS.

**Pull request merged**